x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/conditional/Case.java

```diff
@@ -215,23 +224,20 @@ public boolean foldable() {
             if (condition.condition.foldable() == false) {
                 return false;
             }
-            Boolean b = (Boolean) condition.condition.fold();
-            if (b != null && b) {
+            Object o = condition.condition.fold();
+            if (o instanceof List) {
```

Comment:
I would rewrite this to `Boolean.TRUE.equals(b)` - takes care of nulls, list and other values.